src/fluent/utils.test.ts

```diff
@@ -2,18 +2,56 @@
 // SPDX-FileCopyrightText: 2023-Present The Kubernetes Fluent Client Authors
 
 import { beforeEach, describe, expect, it, jest } from "@jest/globals";
-import { Headers } from "node-fetch";
-
+import * as fs from "fs";
+import { RequestInit } from "node-fetch";
 import { fetch } from "../fetch";
 import { RegisterKind } from "../kinds";
 import { GenericClass } from "../types";
 import { ClusterRole, Ingress, Pod } from "../upstream";
 import { Filters } from "./types";
-import { k8sExec, pathBuilder } from "./utils";
+import { k8sExec, pathBuilder, getHTTPSAgent, getHeaders, getToken } from "./utils";
 
 jest.mock("https");
 jest.mock("../fetch");
 
+describe("getToken", () => {
+  it("should return the token from the service account", async () => {
+    const token = "fake-token";
+    jest.spyOn(fs.promises, "readFile").mockResolvedValue(token as unknown as Buffer);
```

Comment:
Is there a way to do this while avoiding type coercion?

Reply:
What is the strategic advantage of avoiding type coercion when it only returns null?